File: dart/behavioral/observer/guru_example/bin/concrete_observer_b.dart
import 'concrete_subject.dart';
import 'observer.dart';
import 'subject.dart';

class ConcreteObserverB implements Observer {
  @override
  void update(Subject subject) {
    if (subject is ConcreteSubject &&
        (subject.state == 2 || subject.state == 4 || subject.state == 8)) {
      print('ConcreteObserverB');
    }
  }
}

File: typescript/behavioral/observer/guru_example/concreteObserverb.ts
import {Observer} from "./observer";
import {Subject} from "./subject";
import {ConcreteSubject} from "./concreteSubject";

export class ConcreteObserverB implements Observer {
  public update(subject: Subject) {
    if (subject instanceof ConcreteSubject && (subject.state === 1 || subject.state === 3)) {
      console.log('ConcreteObserverB')
    }
  }
}
